README.md

```diff
@@ -170,6 +170,7 @@ _Source:_ [What is Docker](https://www.docker.com/why-docker)
 
 ### Deployment and Infrastructure
 
+- [Pull Dog](https://github.com/apps/pull-dog) - A GitHub app that automatically creates Docker-based test environments for your pull requests, from your docker-compose files.
```

Comment:
* All entries must be ordered alphabetically (from A to Z)
* It should show in the description that it is not open source!
* This is a better section for this entry: https://github.com/veggiemonk/awesome-docker#testing

Reply:
I fixed these points :+1: Thank you.